dispatch.go

```diff
@@ -166,11 +169,11 @@ func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res i
 }
 
 func (d *dispatch) Notify(ctx context.Context, name string, arg interface{}) error {
-	err := d.dispatchMessage(ctx, MethodNotify, name, arg)
-	if err != nil {
-		return err
```

Comment:
I think you need to do something similar here where you select on ctx.Done and errCh, and return/log the appropriate error